primitives/test-utils/src/lib.rs

```diff
@@ -87,23 +94,34 @@ pub fn make_justification_for_header<H: HeaderT>(params: JustificationGeneratorP
 		round,
 		set_id,
 		authorities,
-		depth,
+		mut votes,
 		forks,
 	} = params;
 
 	let (target_hash, target_number) = (header.hash(), *header.number());
 	let mut precommits = vec![];
 	let mut votes_ancestries = vec![];
 
-	assert!(depth != 0, "Can't have a chain of zero length.");
+	assert!(votes >= forks, "Need at least one header per fork.");
 	assert!(
 		forks as usize <= authorities.len(),
 		"If we have more forks than authorities we can't create valid pre-commits for all the forks."
 	);
 
+	// Roughly, how many vote ancestries do we want per fork
+	let target_depth = (votes + forks - 1) / forks;
```

Comment:
Since the divisor (`forks`) is a parameter, this should perhaps be safeguarded against accidental division by '0'.